components/brave_federated_learning/data_stores/ad_notification_timing_data_store.cc

```diff
@@ -0,0 +1,141 @@
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_federated_learning/data_stores/ad_notification_timing_data_store.h"
+
+#include <string>
+#include <utility>
+
+#include "base/bind.h"
+#include "base/guid.h"
+#include "base/json/json_reader.h"
+#include "base/logging.h"
+#include "base/macros.h"
+#include "base/numerics/safe_conversions.h"
+#include "base/strings/stringprintf.h"
+#include "base/time/time.h"
+#include "brave/components/brave_federated_learning/data_stores/data_store.h"
+#include "sql/meta_table.h"
+#include "sql/recovery.h"
+#include "sql/statement.h"
+#include "sql/transaction.h"
+
+namespace {
+
+void BindSampleLogToStatement(
+    const brave::federated_learning::AdNotificationTimingTaskLog&
+        notification_timing_log,
+    sql::Statement* s) {
+  s->BindInt64(0, notification_timing_log.time.ToInternalValue());
+  s->BindString(1, notification_timing_log.locale);
+  s->BindInt(2, notification_timing_log.number_of_tabs);
+  s->BindInt64(3, notification_timing_log.creation_date.ToInternalValue());
+}
+
+}  // namespace
+
+namespace brave {
+
+namespace federated_learning {
+
+// AdNotificationTimingTaskLog ---------------------------------------
+
+AdNotificationTimingTaskLog::AdNotificationTimingTaskLog(
+    const int id,
+    const base::Time& time,
+    const std::string& locale,
+    const int number_of_tabs,
+    const base::Time& creation_date)
+    : id(id),
+      time(time),
+      locale(locale),
+      number_of_tabs(number_of_tabs),
+      creation_date(creation_date) {}
+
+AdNotificationTimingTaskLog::AdNotificationTimingTaskLog(
+    const AdNotificationTimingTaskLog& other) = default;
+
+AdNotificationTimingTaskLog::AdNotificationTimingTaskLog()
+    : id(0),
+      time(base::Time::Now()),
+      locale(std::string()),
+      number_of_tabs(0),
+      creation_date(base::Time::Now()) {}
+
+AdNotificationTimingTaskLog::~AdNotificationTimingTaskLog() {}
+
+// AdNotificationTimingDataStore
+// -----------------------------------------------------
+
+AdNotificationTimingDataStore::AdNotificationTimingDataStore(
+    const base::FilePath& database_path)
+    : DataStore(database_path) {}
+
+bool AdNotificationTimingDataStore::Init(const std::string& task_id,
+                                         const std::string& task_name,
+                                         const int max_number_of_records,
+                                         const int max_retention_days) {
+  return DataStore::Init(task_id, task_name, max_number_of_records,
+                         max_retention_days);
+}
+
+bool AdNotificationTimingDataStore::AddLog(
+    const AdNotificationTimingTaskLog& log) {
+  sql::Statement s(db_.GetUniqueStatement(
+      base::StringPrintf(
+          "INSERT INTO %s (time, locale, number_of_tabs, creation_date) "
+          "VALUES (?,?,?,?)",
+          task_name_.c_str())
```

Comment:
looks like this could lead to sql injection but should be ok as long as `task_name_` is controlled by us